### DISCUSSION TITLE
"IRN: Adblock-Iran" Regional filterlist breakage and unmaintained

### DISCUSSION THREAD

**Discussion by MasterKia** [uBlockOrigin/uAssets#13259] [General]
### Prerequisites

- [X] This is not a [support issue or a question](https://old.reddit.com/r/uBlockOrigin/)
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uAssets/issues) to avoid opening a duplicate issue
- [X] I am running the latest version of uBlock Origin

### I tried to reproduce the issue when...

- [X] uBO is the only extension
- [X] uBO with default lists/settings
- [X] using a new, unmodified browser profile

### URL(s) where the issue occurs

`http://www.ashoora.ir`
`https://bayanbox.ir`
`https://asriran.com` (`http://darunameh.blogfa.com/post/16`)

### Describe the issue

[IRN: Adblock-Iran](https://github.com/farrokhi/adblock-iran) is one of the regional filterlists in uBO.

* `ashoora.ir`:
The `.ir/*/banners/` filter from _Adblock-Iran_ blocks `ashoora.ir/images/banners/Rahyan_410.gif` and `ashoora.ir/images/banners/Behesht_410.gif` at the bottom of the page.
https://user-images.githubusercontent.com/17685483/168642056-69a0aaf3-113d-44ec-9f5b-bbf8f2f2925b.png

* `bayanbox.ir` (Upload Site):
It's a regular **upload** site, where you can upload stuff for free.
The `||bayanbox.ir/view/*/*.gif` filter from _Adblock-Iran_ blocks normal gifs like `https://bayanbox.ir/view/6514224053252251384/Quby-Sticker-2-12.gif` and `https://bayanbox.ir/view/4639030023267517922/Kak-Eunsang.gif` (I forgot which page those gifs were on, but overall this filter is unnecessary).
https://user-images.githubusercontent.com/17685483/168645566-ae6e3617-cf06-4bb8-ae01-c24413ae1caa.png

* `asriran.com`:
It's a **news** site.
The `||cdn.asriran.com^$third-party` filter from _Adblock-Iran_ wrongly blocks `https://cdn.asriran.com/files/fa/news/1398/6/29/1017722_194.jpg`.
https://user-images.githubusercontent.com/17685483/168646580-6d859e0b-3690-4cee-994c-f3068c0a1b1f.png

### Screenshot(s)

`ashoora.ir`:
https://user-images.githubusercontent.com/17685483/168642139-c79f6336-365e-4390-8d46-d0b6c8f8181b.png

### Notes

These filters shouldn't exist:
```
.ir/*/banners/
||bayanbox.ir/view/*/*.gif
||cdn.asriran.com^$third-party
```

### Configuration

<details>

```yaml
uBlock Origin: 1.42.4
Chromium: 101
filterset (summary): 
  network: 83580
  cosmetic: 71753
  scriptlet: 18184
  html: 0
listset (total-discarded, last updated): 
  added: 
    fanboy-cookiemonster: 31879-83, 19m
  default: 
    user-filters: 0-0, never
    easylist: 66428-51, 2d.21h.53m
    easyprivacy: 26943-121, 3d.3h.5m
    plowe-0: 3676-805, 9d.5h.47m
    ublock-abuse: 75-0, 3d.3h.4m
    ublock-badware: 4149-92, 3d.3h.4m
    ublock-filters: 30970-98, 3d.3h.4m
    ublock-privacy: 215-2, 3d.3h.4m
    ublock-quick-fixes: 220-0, 19m
    ublock-unbreak: 1779-41, 2d.20h.18m
    urlhaus-1: 8632-0, 19m
filterset (user): [empty]
modifiedUserSettings: [none]
modifiedHiddenSettings: [none]
supportStats: 
  allReadyAfter: 235 ms (selfie)
  maxAssetCacheWait: 219 ms
```
</details>

**Comment by MasterKia** [comment]
**Side note:**
The current regional Persian filterlist ([adblock-iran](https://github.com/farrokhi/adblock-iran)) isn't maintained (last commit: 5 months ago) unlike other regional lists (For instance: Albania, Indonesian, Indian, Korean, Lithuania and Dutch all have been updated in the last week).
Also the maintainer of _adblock-iran_ has been [nonresponsive to issues](https://github.com/farrokhi/adblock-iran/issues?q=is%3Aopen+is%3Aissue) in the last 5 months (12 open issues, 7 of them submitted by me).

I propose adding the [uBOPa (uBO Parsi/Persian)](https://github.com/nimasaj/uBOPa) filterlist.
uBOPa's maintainer has been responsive to all 46 [issues I've submitted](https://github.com/nimasaj/uBOPa/issues?q=is%3Aissue+author%3AMasterKia+is%3Aclosed) and he updates the list frequently.

Otherwise since _adblock-iran_ is abandoned, I'd have to submit all persian regional issues here.

**Comment by JobcenterTycoon** [comment]
We know this list already but the list contains many problematic filters like `*##*[class*=" adv "]`

**Comment by MasterKia** [comment]
@nimasaj

**Comment by MasterKia** [comment]
> We know this list already but the list contains many problematic filters like `*##*[class*=" adv "]`

If he removes problematic ~~(mostly cosmetic)~~ filters, is there a chance of adding this list?

**Comment by Yuki2718** [comment]
No, not only cosmetic. We really can't accept such way too generic rules like `*-ads-^$script` `||ad0*` `*_adv-^$image`. Note `||ad0*` is not even tokenizable.

**Comment by MasterKia** [comment]
I assume he'd be willing to get rid of those filters you mentioned and only include network filters and non-generic cosmetic rules.

The Adblock-Iran has some redundant or problematic filters too
Some redundant filters on top of my head:
https://github.com/farrokhi/adblock-iran/blob/master/filter.txt#L684-L703
https://github.com/farrokhi/adblock-iran/blob/master/filter.txt#L1263
https://github.com/farrokhi/adblock-iran/blob/master/filter.txt#L280

As for problematic filters, for instance it has 183 ".gif" filters for no apparent reason.
https://user-images.githubusercontent.com/17685483/168655887-bfaea245-f818-4a59-b70c-efe318e0d8be.png

Can't a website upload anything other than ads as a gif?

**Comment by JobcenterTycoon** [comment]
.gif can be animated so it get most used for ads, other picture formats like .png have a higher quality so it get used for the content. thats why.

**Comment by MasterKia** [comment]
Fair enough.
If _Adblock-Iran_ is gonna stay unmaintained, do you suggest merging future Persian filters into **uBlock filters** or **EasyList Ads/Privacy**?

Although if uBOPa has a chance of getting added to uBO I'm sure uBOPa's maintainer will remove those problematic filters.

**Comment by krystian3w** [comment]
Maybe possible fork list and enable actively maintaing - if older repo can't be move to new hands (the best create organization to fallback legacy links) or enable new admin user with high privileges.

PS. AdGuard now need ~10 stars on repo and ~10 commits at month to add or change exist Iran list to new.
https://github.com/AdguardTeam/FiltersRegistry/blob/master/filters/ThirdParty/filter_235_adblock-iran/metadata.json

**Reply by MasterKia** [reply]
> Maybe possible fork list and enable actively maintain

Good idea, me and @nimasaj can maintain it.

There are 3 persian filterlist that I know of:
https://github.com/farrokhi/adblock-iran
https://github.com/SlashArash/adblockfa
https://github.com/nimasaj/uBOPa

We could combine these into one new list and maintain it from there.

uBOPa has 15 stars and ~500 commits.

**Reply by MasterKia** [reply]
~~I'm combining those 3 lists into one list and removing redundant filters.~~ [Currently on hold](https://github.com/uBlockOrigin/uAssets/discussions/13259#discussioncomment-2767569)

**Comment by nimasaj** [comment]
I find some of the comments in this topic irrelevant or unclear. The filters are either specific to a domain or generic, acting as regular expressions, when a pattern used by a number of websites is found. 
The generic filters, besides their benefits, might introduce some wrong triggers, and that's one of the reasons of developing/maintaining the list by adjusting the filters, to either narrow down and/or remove a filter or add exceptions for wrongly affected domains, as it could be seen in other filter lists too. uBOPa is not an exception here and if there are reports of a wrongly blocked element/address for a generic filter, a similar approach will be taken, by either adding an exception, or narrowing down or removing the filter. 

The filters in uBOPa are developed based on the keywords used for ad elements/addresses in Iran. And even though some of the filters might not look good/acceptable to you, but the actual case is different and they're performing as they should do. 

A regional filter lists is mostly used by the people of that region and if there are issues, it could be reported to the list maintainer. Again, I don't see why there is resistance from people who are not intended to use that list.
I don't want to go in a discussion of proving how bad/wrong some of the filters are in some of the lists that are already used or provided by default by uBO as there can be many examples. 

If the goal of having the regional list is to serve the people with most updated and maintained list, my recommendation is to have a trial phase to have the feedback from the users.

**Comment by Yuki2718** [comment]
@nimasaj One question: are you willing to cooperate with us? Frankly, a number of rules in your list break uBO's effort for it to be *efficient* blocker. You probably just didn't know and it's not my intention to attack you or your list, but in any case for the list to be considered at least following must be addressed or improved:

1. Remove all unneeded wildcards.

ABP syntax automatically assumes wildcard around each rule, so `baner&` matches `/etc/baner&param=` without explicit wildcard. Here uBO-specific [exception](https://github.com/uBlockOrigin/uBlock-issues/issues/954#issuecomment-602041890) is that on uBO it matches `?baner&` but does not `abaner&` by itself because here token from the request is `abaner` while token from the rule is `baner`. This is the key mechanism of uBO and adding wildcard `*baner&` breaks this, now it matches `abaner&` but the number of such rules directly affects performance while the number of rules like `baner&` not.

`*##*[alt="ads"]` too (`*` after `#` is simply redundant and not needed), `*` before `#` makes the rule [specific generic](https://github.com/gorhill/uBlock/wiki/Static-filter-syntax#specific-generic) which should be used only if necessary and there's no other way. Also note `###id` or `##.class` is far prefered to `##[class*="something"]`, we're even [discussing](https://github.com/uBlockOrigin/uBlock-issues/discussions/1998) whether we should disable rules like the latter by default. I also noticed quite many rules end with `^` if not `*`, but `^` is a separator matching anything that isn't alphanumerical, `_-.%` or the end-of-lines, I guess most of those `^` are not needed either.

2. Remove or replace unduly broad generic rules

I read your argument but any regional list can well be used by user in other region, and even in the intended region people visit international sites. The difference with other stock lists are the number and extent of such rules. User may report issues not only to your repo but to uAssets too, here's gorhill's opinion about generic rule which I fully agree with: https://github.com/uBlockOrigin/uBlock-issues/issues/738#issuecomment-536180398 Which rules you regard as potentially too broad is certainly your choice, IIRC some websites use class name `adv` for legitimate contents to represent "adventure", but there will be many such cases if inspect further. Plus they can be a fascinating bait for anti-adb. You can probably have discussion with @MasterKia about this.

**Reply by krystian3w** [reply]
End of path will be `|` in network filtering

**Reply by MasterKia** [reply]
@Yuki2718  I don't disagree with your points, because I've come across some sites where `*##*[class$="..."]` broke something and I've repeatedly pointed out the problem of those generic cosmetic rules in several [issues](https://github.com/nimasaj/uBOPa/issues?q=breakage).

As for the current state of the Persian filterlist, I hope @nimasaj comes to a conclusion and the uBOPa list would have a chance to get considered.

Otherwise, I have an [idea](https://github.com/uBlockOrigin/uAssets/discussions/13259#discussioncomment-2764334) to combine the three well known Persian filterlist into one list, removing redundant/dead filters and the generic network/cosmetic rules and only including site-specific filters (`example.ir/ads-link` - `example.ir##.adsBlock - example.ir##+js(...)`).
Filters that are already covered by uBO stock lists and EasyList Ads/Privacy wouldn't be included.

Would such list have a possibility of being considered if I continue expanding it?

**Reply by Yuki2718** [reply]
That's not what I can answer without internal discussion. I proposed some changes and waiting answer from the author as you see, if he can accept this will be best - we don't know if your new combined list will be well-maintained. But one point in your comment I didn't pointed out is
> Filters that are already covered by uBO stock lists and EasyList Ads/Privacy wouldn't be included.

Sure good compatibility with defualt lists is preferable but TBH some regional lists have quite redundancy with EL.

**Comment by MasterKia** [comment]
More breakage
`https://article.tebyan.net/306262/%D8%AD%D8%A7%D9%84%D8%A7-%D8%A7%DA%AF%D8%B1-%D9%85%DB%8C-%D8%AE%D9%88%D8%A7%D9%87%DB%8C%D8%AF-%D9%BE%D9%81%DA%A9-%D8%A8%D8%AE%D9%88%D8%B1%DB%8C%D8%AF`

`tebyan.net##.theiaStickySidebar`
![image](https://user-images.githubusercontent.com/17685483/169347038-f591c715-8317-4ff3-a953-6624446a12df.png)

https://user-images.githubusercontent.com/17685483/169347205-f07680a6-fd82-44b6-a598-5cf66bfa4333.png

**Comment by mapx-** [comment]
You should prepare a PR for all these breakages

**Reply by MasterKia** [reply]
https://github.com/uBlockOrigin/uAssets/pull/13308

**Comment by MasterKia** [comment]
> That's not what I can answer without internal discussion. I proposed some changes and waiting answer from the author as you see, if he can accept this will be best - we don't know if your new combined list will be well-maintained.

Well since it's been a week so I assume the author isn't interested.

I'm still volunteering to _maintain_ a new combined Persian filterlist (Adblock-Iran + uBOPa + Adblock Farsi).
@Yuki2718 May I know whether you discussed that internally?

**Reply by Yuki2718** [reply]
Discussing

**Reply by Yuki2718** [reply]
@MasterKia Got only positive response. Once your new list have got matured and is ready to be included, tell us again in some way.

**Reply by MasterKia** [reply]
Sure thing, thank you for your time.

**Comment by MasterKia** [comment]
PersianBlocker:
https://raw.githubusercontent.com/MasterKia/PersianBlocker/main/PersianBlocker.txt

I've rewritten filters of adblock-iran from scratch (per site) and I've removed dead sites/filters and added some missing ads/tracker filters for each site.
So as of now, this list should hopefully be a _drop-in_ _**improved**_ replacement for adblock-iran.

In the next days I'll bring new sites/filters from uBOPa and Adblock Farsi to increase the list's coverage and usefulness.

If the list is acceptable, should I submit a PR for https://github.com/gorhill/uBlock/blob/master/assets/assets.json?

Also there were some breakages coming from EasyList/uAssest which I've marked with "TODO" and I'm going to submit issues for them.

**Reply by okiehsch** [reply]
> If the list is acceptable, should I submit a PR for https://github.com/gorhill/uBlock/blob/master/assets/assets.json?

The only one with write access in that repo is gorhill and right now he is not available, so you can open a pull request but it could take some time until it is either accepted or declined.

**Reply by MasterKia** [reply]
Alright, I understand.

> you can open a pull request until it is either accepted or declined.

I was asking for some kind of **approval** on the list from [one of the] maintaines _before_ I go and open a pull request since @Yuki2718 told me to inform you when the list is ready and matured.

**Reply by okiehsch** [reply]
I am one of those maintainers and I am in favour of replacing the regional list with yours and I said as much in the internal discussion.
The final decision is gorhills to make and he has not chimed in as of now.

**Reply by MasterKia** [reply]
Thank you, that's what I wanted to know, I'll open a pull request and wait to see what happens.

**Comment by MasterKia** [comment]
"[An owner of this repository has limited the ability to open a pull request to users that have contributed to this repository in the past.](https://github.com/gorhill/uBlock/pulls)"

I'd be grateful if one of the maintainers could open a pull request on my behalf, otherwise I guess we should wait until @gorhill adds the list.

https://github.com/MasterKia/uBlock/commit/fa838ff39bc0c01beb314093803b72c9635c5168

**Reply by Yuki2718** [reply]
Just wait for his decision with commit in your branch kept.

**Comment by Yuki2718** [comment]
BTW gorhill seems to be silent in all activities for a few weeks. Probably you have to be patient.

**Reply by MasterKia** [reply]
May I know what is _your_ opinion on this?
`https://github.com/easylist/easylist/issues/12147`
[EasyList Coverage](
https://easylist.to/pages/policy.html) states `Languages for which there are no supplements may have issues resolved in EasyList` and _**currently**_ there is no supplement for Persian but the maintainer arbitrarily rejected fixing those breakages.

I'm opening a pull request for uBlock Unbreak.

**Reply by ghost** [reply]
Does AdGuard know about the `PersianBlocker` filter list, @Yuki2718 ?

**Reply by Yuki2718** [reply]
I think breakage caused by EL should be fixed in EL as far as possible and actually opened many such PRs most of which were merged. If I fix in regional list (Japanese filter in my case), it's either to save my time or because there are too many cases. Maybe you put too many in one PR but I don't know, it's their decision. PR in uAssets will be merged after check until IRN list get replaced by yours.

**Reply by MasterKia** [reply]
> Maybe you put too many in one PR

I agree, I thought of creating several (maybe 18 issues for each site) at first but then I assumed that it would be more irritating than one creating one issue.

**Reply by MasterKia** [reply]
> Does AdGuard know about the `PersianBlocker` filter list?

@Branzeturi 
`https://github.com/AdguardTeam/FiltersRegistry/issues/617`

**Comment by ghost** [comment]
Also, perhaps https://github.com/easylist/easylist/issues/12147 may be useful with decision making regarding the `PersianBlocker` filter list -- or adding filters to the `uBO unbreak` filter list

**Comment by MasterKia** [comment]
@gorhill https://github.com/uBlockOrigin/uAssets/discussions/13259#discussioncomment-2840965

**Reply by gorhill** [reply]
https://github.com/gorhill/uBlock/commit/33b839fdd03f74689df3ee2b5c25a06435b350e0

**Reply by MasterKia** [reply]
Users who have _already_ enabled Adblock-Iran, would eventually (after filter updates) get the PersianBlocker list instead?
So that existing users of Adblock-Iran would receive the new list eventually.

**Reply by ghost** [reply]
Yes. Purge->update->refresh the dashboard, you will see new list name.

**Reply by farrokhi** [reply]
> [gorhill/uBlock@33b839f](https://github.com/gorhill/uBlock/commit/33b839fdd03f74689df3ee2b5c25a06435b350e0)

All these efforts could have been reported to Adblock-Iran project. Entries in block list can go broken or stale.

**Reply by Yuki2718** [reply]
> All these efforts could have been reported to Adblock-Iran project. Entries in block list can go broken or stale.

How you can say so when the last update was 18 Dec 2021.

**Reply by farrokhi** [reply]
@gorhill I don't really mind about this change. But this is clearly a move to whitelist ads some well known sites (i.e. soft98, asriran, etc). They have been trying this for a long time, and clearly, advertisers always win ;-)

**Reply by mapx-** [reply]
> But this is clearly a move to whitelist ads some well known sites

It is not enough to state something, you must also prove that fact, documented examples ?

**Reply by farrokhi** [reply]
This is one: https://github.com/farrokhi/adblock-iran/issues/71

**Reply by Yuki2718** [reply]
Describe what *ads* are allowed by the change. OTOH I can trivially confirm `||cdn.asriran.com^$third-party` is really a bad filter, blocking many *non-ads*. Note *ads* != self-promo.

**Reply by farrokhi** [reply]
> Describe what _ads_ are allowed by the change. OTOH I can trivially confirm `||cdn.asriran.com^$third-party` is really a bad filter, blocking many _non-ads_. Note _ads_ != self-promo.

When used as a third-party, this site only serves ads.

**Reply by Yuki2718** [reply]
So you say this is ad (not even clickable)?

![blogfa](https://user-images.githubusercontent.com/58900598/172436516-25c882cc-ed15-4929-ad72-3a0ddb3d6338.png)

**Reply by okiehsch** [reply]
The picture is obviously related to the text besides it and that translated text does not strike me as advertising.

"It gets to the point where the loss of this connection wears out the soul and body of individuals.
Usually, people who intend to abuse emotionally behave differently when confronted with the other party seriously for the relationship, in some cases, mutual expression of problem, rejection, accusation, anger, and in others, resorting to promise and promise. They apologize."

**Reply by Yuki2718** [reply]
False positive and misrecognition happens, that's okay. What led to the change was 6 months inactivity without any sign of being alive (e.g. response to issues). We've recently got upset as List-FR had been inactive for a month, even discussed for the worst case.

**Reply by MasterKia** [reply]
> All these efforts could have been reported to Adblock-Iran project

![image](https://user-images.githubusercontent.com/17685483/172448424-3f72a5a3-d393-4c1e-9329-778f68d00413.png)

**Reply by MasterKia** [reply]
And two of my other issues (`eghtesadonline.com` ads/`ashoora.ir` breakage) have been closed tonight without any answer.

**Reply by MasterKia** [reply]
> whitelist ads some well known sites (i.e. soft98, asriran, etc).

Soft98:
https://github.com/MasterKia/PersianBlocker/blob/main/PersianBlocker.txt#L1504-L1516

Asriran: 
https://github.com/MasterKia/PersianBlocker/blob/main/PersianBlocker.txt#L408-L413

To be specific, I created [this](https://github.com/uBlockOrigin/uAssets/issues/10883
) issue for soft98 and pushed for  over a month to achieve better/effective filters despite soft98's monstrous anti-adblock.
The soft98 cat and mouse game even got to the point where I proposed adding a clever userscript written by @DRSDavidSoft to uBO's stock scriptlets just to counter soft98's pathetic attempts, which gorhill reasonably declined (because the userscript was huge).

Considering all that, how could I be the _one_ who "clearly made a move to whitelist ads of well known sites"?

**Reply by MasterKia** [reply]
> This is one [attempt to whitelist ads from soft98]: [farrokhi/adblock-iran#71](https://github.com/farrokhi/adblock-iran/issues/71)

Quotes from https://github.com/uBlockOrigin/uAssets/issues/10883:
@MasterKia: 
> Some of these filters from the [uBlock filters](https://github.com/uBlockOrigin/uAssets/blob/master/filters/filters.txt#L10765) aren't working anymore (similar filters can be found in the [IRN: Adblock-Iran](https://github.com/farrokhi/adblock-iran/blob/master/filter.txt#L677) list from @farrokhi)

@DRSDavidSoft:
> I tried pushing some filters to the known iranian filters (https://github.com/farrokhi/adblock-iran/blob/master/filter.txt) and that didn't quite work (remnants are still present), so I went ahead and wrote my userscript.

So:

> this is clearly a move to whitelist ads some well known sites (i.e. soft98)

It seems _you_ were the one reluctant to address the issues of soft98, why?

**Reply by farrokhi** [reply]
> False positive and misrecognition happens, that's okay. What led to the change was 6 months inactivity without any sign of being alive (e.g. response to issues). We've recently got upset as List-FR had been inactive for a month, even discussed for the worst case.

This is a list that I update when I run into new ads, as well as based on user input. I regularly receive claims of sites being broken without additional details or proof. The breakage may have various reasons and without enough technical details, I cannot verify the claim. Contributors want to push their changes without proper testing, etc, and I need to moderate everything properly. As a maintainer you are not supposed to let every change land into the repo. In this case, this user created a new repo and started spamming on the comments to promote the new list. Contributors should be helpful, not pushy. All these effort on spamming and self promotion could have gone into creating proper bug reports.
There are also numerous examples of proper reports and contributions being merged into the repo (as you can see in git history).

**Reply by MasterKia** [reply]
> creating proper bug reports

I'll let the pictures speak for themselves:

https://user-images.githubusercontent.com/17685483/172455152-0ed624d6-7d02-4b5b-9b9a-c66cd0ba51a5.png

https://user-images.githubusercontent.com/17685483/172455192-1174dad1-13ac-4c9e-8c39-cb8723634709.png

These issues haven't been answered in 6 months, despite being "proper bug reports".
